### ISSUE TITLE
Increase amount of programming languages shown from Top 1 to Top 3.

### ISSUE THREAD

**Issue by CSfromCS** [DeepSourceCorp/good-first-issue#2246]
Most of the repos have larger codebases and would benefit in including secondary languages. 

It is also harder to filter for supplementary languages that are often never the main language used.

**Comment by Nitish-Naik** [comment]
@CSfromCS can you elaborate more what do you want to tell

**Comment by fgiaquintaruiz** [comment]
yeah, also I'm trying to contribute to the vue js part and there is not even a filter for it hehe

**Comment by bnspopi** [comment]
Hi, I’d like to work on adding secondary language filtering support.
Could you confirm the expected behavior and where the filtering logic currently resides?

**Comment by rootaashu** [comment]
To display Top 3 languages instead of just Top 1:
1. We might fetch the language statistics (GitHub API returns a language breakdown).
2. Then sort languages by size and show the top 3.

Could you confirm how the UI should show it (e.g., comma-separated, badges)?

**Comment by ibrhim2344** [comment]
Hi, I'd like to work on this issue. Can you assign it to me?

**Comment by Tecnic-debug** [comment]
Hey I would like to solve the issue

**Comment by KyriakosVerveridis** [comment]
Hi, I'd like to work on this issue.

**Comment by Gradient-learner** [comment]
Hi, I would like to work on this issue under GSSoC. Please assign it to me.

**Comment by Darshijain14** [comment]
Hi, I’d like to work on this issue. this is my first contribution

**Comment by sanachishtti-glitch** [comment]
Hi, I would like to work on this issue. Can I take it?

**Comment by Afreenshaik07** [comment]
Hi, I would like to work on this issue. This will be my first contribution. Could you please assign it to me?

**Comment by Varadraj2327** [comment]
Hi, this is my first time working on some issues, so can i take it?

**Comment by Mariya115** [comment]
Hi! I'd like to work on this issue. I noticed the current implementation only stores `repository.language` in `generated.json`. Before I start implementing support for the top 3 languages, could you confirm the expected UI format (comma-separated list, badges, etc.) and whether language breakdown data should be fetched during data generation?

**Comment by fragan7dsouza** [comment]
Hello, i will be working on this issue as my first contribution

**Comment by supriya1411** [comment]
Hi! I'm interested in working on this issue. I have reviewed the requirements and I'm ready to start. If no one is currently working on it, could you please assign it to me? Thank you!

**Comment by nishanth-28-hub** [comment]
Hello!  I am interested in working on this issue. This would be my first contribution, @CSfromCS could you please assign it to me

**Comment by sajiv8** [comment]
Hi! I'm interested in working on this issue. I have reviewed the requirements and I'm ready to start. If no one is currently working on it, could you please assign it to me?
